src/components/SlateEditor/plugins/related/RelatedArticle.tsx

```diff
@@ -1,37 +1,46 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+// @ts-ignore
 import { RelatedArticle as RelatedArticleUI } from '@ndla/ui';
 import { useTranslation } from 'react-i18next';
 import { convertFieldWithFallback } from '../../../../util/convertFieldWithFallback';
 import { mapping as iconMapping } from '../../utils/relatedArticleMapping';
 import { urlDomain } from '../../../../util/htmlHelpers';
 import { toEditArticle } from '../../../../util/routeHelpers';
+import { RelatedArticleType } from './RelatedArticleBox';
 
-const resourceTypeProps = (item, numberInList) => {
-  const resourceType = item.resource
-    ? item.resource.length &&
-      item.resource[0].resourceTypes &&
-      item.resource[0].resourceTypes.find(
-        resourceType => iconMapping(numberInList)[resourceType.id],
-      )
-    : { id: item.id }; // if no resource it is external article
+const resourceTypeProps = (item: RelatedArticleType, numberInList?: number) => {
```

Comment:
Kunne gjort denne litt enklere å lese, muligens?
Legger ved et eksempel som jeg ikke har testet

const resourceTypes = 'resource' in item ? item?.resource[0]?.resourceTypes : undefined;
const resourceTypeId = resourceTypes?.find(rt => iconMapping(numberInList)[rt.id])?.id ?? item.id;
const mappingType = resourceTypeId ?? 'default';
return iconMapping(numberInList)[mappingType];